src/Plugin/Styleguide/searchStyleguide.php

```diff
@@ -0,0 +1,253 @@
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   * @param \Drupal\styleguide\GeneratorInterface $styleguide_generator
+   * @param ThemeManagerInterface $theme_manager
+   * @param \Drupal\Core\Form\FormBuilder $form_builder
+   * @param ModuleHandlerInterface $module_handler
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *
+   * @internal param \Drupal\Core\Breadcrumb\ChainBreadcrumbBuilderInterface $breadcrumb
+   * @internal param \Drupal\styleguide\GeneratorInterface $generator
+   */
+
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, GeneratorInterface $styleguide_generator, ThemeManagerInterface $theme_manager, ModuleHandlerInterface $module_handler, FormBuilder $form_builder, AccountInterface $current_user, EntityManagerInterface $entity_manager, RendererInterface $renderer) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->generator = $styleguide_generator;
+    $this->themeManager = $theme_manager;
+    $this->moduleHandler = $module_handler;
+    $this->formBuilder = $form_builder;
+    $this->currentUser = $current_user;
+    $this->entityManager = $entity_manager;
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('styleguide.generator'),
+      $container->get('theme.manager'),
+      $container->get('module_handler'),
+      $container->get('form_builder'),
+      $container->get('current_user'),
+      $container->get('entity.manager'),
+      $container->get('renderer')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function items() {
+    $items = [];
+    if ($this->moduleHandler->moduleExists('search')) {
+      $items['search_block_form'] = [
+        'title' => $this->t('Search block form'),
+        'content' => $this->formBuilder->getForm('Drupal\search\Form\SearchBlockForm'),
+        'group' => $this->t('Search'),
+      ];
+
+      $items['search_no_results'] = [
+        'title' => $this->t('Search, no results'),
+        'content' => [
+          '#theme' => 'item_list__search_results',
+          '#items' => [],
+          '#empty' => [
+            '#markup' => '<h3>' . $this->t('Your search yielded no results.') . '</h3>',
+          ],
+        ],
+        'group' => $this->t('Search')
+      ];
+
+      // Generate sample results.
+      $this->searchResults($items);
+    }
+
+    return $items;
+  }
+
+  /**
+   * Generate fake search results.
+   *
+   * @param $items
+   */
+  private function searchResults(&$items) {
+    $results = [];
+    if (\Drupal::hasService('plugin.manager.search')) {
```

Comment:
I'd better revert to previous one but made `SearchPluginManager $search_manager = NULL` in `__construct()`
Also this check better to make in `create()`

This way plugin points that the service is optional